src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -11438,6 +11438,11 @@ static string Invoke(Action a)
                     var synthesizedType = module.GlobalNamespace.GetTypeMember("<>z__ReadOnlyArray");
                     Assert.Equal("<>z__ReadOnlyArray<T>", synthesizedType.ToTestDisplayString());
                     Assert.Equal("<>z__ReadOnlyArray`1", synthesizedType.MetadataName);
+
+                    Assert.Contains(synthesizedType.GetAttributes(),
```

Comment:
>Assert [](http://example.com/codeflow?start=20&length=6)

Please also test `<>z__ReadOnlySingleElementList<T>` and `<>z__ReadOnlyList`.

For instance, consider splitting this out to a new test and testing on the types from:
```csharp
IEnumerable<int> x = [1];
IEnumerable<int> y = [2, 3];
IEnumerable<int> z = [..x];
```